### ISSUE TITLE
no-unused-vars confused by try-catch

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#1575]
It appears that `no-unused-vars` gets confused when a variable is first used in a `try-catch`. I have the following code in Espree:

``` js
function testRegExp(pattern, flags) {
    var value;

    // First, detect invalid regular expressions.
    try {
        value = new RegExp(tmp);
    } catch (e) {
        throwError({}, Messages.InvalidRegExp);
    }
}
```

And I get the following using 0.10.2:

```
espree.js
  805:8   error  value is defined but never used  no-unused-vars
```

I've been able to consistently able to reproduce this using this same pattern.
